web-platform/website/src/services/ExtensionsEventService.ts

```diff
@@ -45,7 +49,70 @@ function unsubscribeFromEvents() {
   );
 }
 
-function onUserTokenRequested() {}
+async function onUserTokenRequested(e: Event) {
+  const customEvent = e as CustomEvent;
+
+  const detail =
+    customEvent && customEvent.detail && JSON.parse(customEvent.detail);
+
+  const error =
+    (!customEvent && "Invalid complete signup event.") ||
+    (!customEvent.type && "Invalid complete signup event type.") ||
+    (!detail && "Invalid complete signup event detail.") ||
+    (!detail.studyId && "Invalid complete signup event study id.") ||
+    "";
+
+  assert(!error, error);
+
+  const { studyId } = JSON.parse(customEvent.detail);
+
+  const rallyToken = await getRallyTokenForLoggedInUser(studyId);
+
+  window.dispatchEvent(
+    // Each study needs its own token. Send to content script.
+    new CustomEvent("rally-sdk.complete-signup-response", {
```

Comment:
We should figure out how to share these types: https://github.com/mozilla-rally/rally/blob/6142b3883c5817c142c4da0b78df8036b83faa94/extensions/sdk/src/WebMessages.ts

Reply:
Actually we should move this to common/types and consume these from SDK. SDK is meant to be implemented by extensions and not web platform.